wagtail/api/v2/views.py

```diff
@@ -440,8 +441,16 @@ def get_base_queryset(self):
         This is used as the base for get_queryset and is also used to find the
         parent pages when using the child_of and descendant_of filters as well.
         """
-        # Get live pages that are not in a private section
-        queryset = Page.objects.all().public().live()
+        # Get all live pages
+        queryset = Page.objects.all().live()
+        # if user is not authenticated return only public pages
+        if not self.request.user.is_authenticated:
+            queryset &= queryset.public()
+        else:  # get pages that are accessible to logged-in users or users in specific groups
+            queryset &= queryset.filter(Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN)
+                                        | Q(view_restrictions__groups__in=self.request.user.groups.all())
+                                        | Q(view_restrictions__restriction_type=None))
```

Comment:
Indentation here looks a little off
```suggestion
            queryset &= queryset.filter(
                Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN) |
                Q(view_restrictions__groups__in=self.request.user.groups.all()) |
                Q(view_restrictions__restriction_type=None)
            )
```

Reply:
You're right, but your suggestion broke this rule https://www.flake8rules.com/rules/W504.html, I think it should be like that ->
```suggestion
          queryset &= queryset.filter(
                Q(view_restrictions__restriction_type=PageViewRestriction.LOGIN)
                | Q(view_restrictions__groups__in=self.request.user.groups.all())
                | Q(view_restrictions__restriction_type=None)
            )
```